CodeSignalJS/reverseInParentheses.ts

```diff
@@ -4,6 +4,42 @@
 // Ex input = "(bar)" => "rab"
 // Ex input = "foo(bar(baz))blim" => "foobazrabblim"
 
-function reverseInParentheses(s: string): string { return "string"}
+function reverseInParentheses(s: string) { 
+    let characterArray = s.split('');
+
+    const parenIndexes = (characterArray) : {start:number | undefined, end:number | undefined} => {
+        let startParen : number | undefined = undefined;
+        let endParen: number | undefined = undefined;
+        for(let i = 0; i < characterArray.length ; i++){
+            let el = characterArray[i]
+            if(el == '('){
+                startParen = i;
+            } else if (el == ')') {
+                endParen = i
+                return {start: startParen,end: endParen}
+            } 
+        } 
+        return {start: startParen,end: endParen}
+    }
+    
+    let startParen = parenIndexes(characterArray).start;
+    let endParen = parenIndexes(characterArray).end;
```

Comment:
note that these two lines will force parenIndexes to run twice in a row, if this were a more expensive method to call (for example one that invokes an api) this would not be ideal, how can we ensure it's only called once?